torch/_inductor/graph.py

```diff
@@ -353,6 +353,32 @@ def codegen(self):
 
         self.wrapper_code = WrapperCodeGen()
         self.scheduler = Scheduler(self.buffers)
+        from .utils import sympy_product
+        def get_read_write_buffers_sizes(node):
+            reads = set([dep.name for dep in node.read_writes.reads])
+            writes = set([dep.name for dep in node.read_writes.writes])
+            total_numel = 0
+            for name in set(reads + writes):
+                if name in self.scheduler.name_to_node:
+                    buf = self.scheduler.name_to_node[name].node
+                    if isinstance(buf.layout, ir.MultiOutputLayout):
+                        continue
+                    numel = V.graph.sizevars.size_hint(sympy_product(buf.get_size()))
+                    print(buf)
+                    print(name, numel)
+                    total_numel += numel
+                elif name in V.graph.graph_inputs:
+                    buf = V.graph.graph_inputs[name]
+                    numel = V.graph.sizevars.size_hint(sympy_product(buf.get_size()))
+                    print(name, numel)
+                    total_numel += numel
+            return total_numel
+        total_numel = 0
+        for node in self.scheduler.nodes:
+            numel = get_read_write_buffers_sizes(node)
```

Comment:
that's a nice place to put it in :-)

Reply:
I'm probably just going to make it a separate method on Graph.